plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java

```diff
@@ -178,27 +177,27 @@ public void sinkFinished(ExchangeSinkHandle handle, int taskAttemptId)
             FileSystemExchangeSinkHandle sinkHandle = (FileSystemExchangeSinkHandle) handle;
             finishedSinks.putIfAbsent(sinkHandle.getPartitionId(), taskAttemptId);
         }
-        checkInputReady();
     }
 
-    private void checkInputReady()
+    @Override
+    public void allRequiredSinksFinished()
     {
         verify(!Thread.holdsLock(this));
-        ListenableFuture<List<ExchangeSourceHandle>> exchangeSourceHandlesCreationFuture = null;
+        ListenableFuture<List<ExchangeSourceHandle>> exchangeSourceHandlesCreationFuture;
         synchronized (this) {
             if (exchangeSourceHandlesCreationStarted) {
                 return;
             }
-            if (noMoreSinks && finishedSinks.keySet().containsAll(allSinks)) {
-                // input is ready, create exchange source handles
-                exchangeSourceHandlesCreationStarted = true;
-                exchangeSourceHandlesCreationFuture = stats.getCreateExchangeSourceHandles().record(this::createExchangeSourceHandles);
-                exchangeSourceHandlesFuture.whenComplete((value, failure) -> {
-                    if (exchangeSourceHandlesFuture.isCancelled()) {
-                        exchangeSourceHandlesFuture.cancel(true);
-                    }
-                });
-            }
+            verify(noMoreSinks, "noMoreSinks is expected to be set");
+            verify(finishedSinks.keySet().containsAll(allSinks), "all sinks are expected to be finished");
```

Comment:
why? Some sinks may still be written, right?

Reply:
Currently we don't split tasks and don't "forget" entire partitions. Once we have this functionally we will remove this check. For now I wanted to keep it in place for verification purposes.